clippy_lints does not compile:  non-primitive cast: `rustc_target::abi::Size` as `usize`

```
$ rustup update                          
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
info: checking for self-updates

   stable-x86_64-unknown-linux-gnu unchanged - rustc 1.26.0 (a77568041 2018-05-07)
  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.28.0-nightly (a3085756e 2018-05-19)

$ cargo +nightly install --force clippy

error[E0605]: non-primitive cast: `rustc_target::abi::Size` as `usize`
   --> /home/brain/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.202/src/consts.rs:435:34
    |
435 |                     let offset = ptr.offset as usize;
    |                                  ^^^^^^^^^^^^^^^^^^^
    |
    = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait

error: aborting due to previous error

For more information about this error, try `rustc --explain E0605`.
error: failed to compile `clippy v0.0.202`, intermediate artifacts can be found at `/tmp/cargo-install87BDnW`

--- a/clippy_lints/src/consts.rs
+++ b/clippy_lints/src/consts.rs
@@ -432,7 +432,7 @@ pub fn miri_to_const<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, result: &ty::Const<'
                         .interpret_interner
                         .get_alloc(ptr.alloc_id)
                         .unwrap();
-                    let offset = ptr.offset as usize;
+                    let offset = ptr.offset.bytes() as usize;
                     let n = n as usize;
                     String::from_utf8(alloc.bytes[offset..(offset + n)].to_owned()).ok().map(Constant::Str)
                 },

